File: iteration2/LaboratorySection.java
package iteration2;

/**
 * The LaboratorySection class represents a course with a laboratory offered by an educational institution in different departments.
 * It contains extra information about the laboratorySectionCode, capacity, assistant and numberOfStudents,hour,day of the section.
 */
public class LaboratorySection {
    private String laboratorySectionCode;
    private int capacity;
    private Assistant assistant;
    private int numberOfStudents;
    private int hour;
    private String day;


    /**
     * Creates a new LaboratorySection object with the given parameters
     * @param laboratorySectionCode      the section's code
     * @param capacity                   the section's capacity
     * @param hour                       the section's hour
     * @param day                        the section's day
     */

   public LaboratorySection(String laboratorySectionCode, int capacity, int hour, String day) {
        this.laboratorySectionCode = laboratorySectionCode;
        this.capacity = capacity;
        this.hour = hour;
        this.day = day;
    }

    public String getLaboratorySectionCode() {
        return laboratorySectionCode;
    }

    public void setAssistant(Assistant assistant) {
        this.assistant = assistant;
    }

    public String getDay() {
        return day;
    }

    public int getHour() {
        return hour;
    }

    public int getCapacity() {
        return capacity;
    }

    public int getNumberOfStudents() {
        return numberOfStudents;
    }

    public void setNumberOfStudents(int numberOfStudents) {
        this.numberOfStudents = numberOfStudents;
    }

  
}

File: iteration3/LaboratorySection.py
class LaboratorySection:

    def __init__(self, laboratorySectionCode, capacity, hour, day):
        self.__laboratorySectionCode = laboratorySectionCode
        self.__capacity = capacity
        self.__hour = hour
        self.__day = day
        self.__assistant = None
        self.__numberOfStudents = 0

    def getLaboratorySectionCode(self):
        return self.__laboratorySectionCode

    def setAssistant(self, assistant):
        self.__assistant = assistant

    def getDay(self):
        return self.__day

    def getHour(self):
        return self.__hour

    def getCapacity(self):
        return self.__capacity

    def getNumberOfStudents(self):
        return self.__numberOfStudents

    def setNumberOfStudents(self, numberOfStudents):
        self.__numberOfStudents = numberOfStudents
